uspace/app/taskbar/wndlist.c

```diff
@@ -716,13 +716,27 @@ errno_t wndlist_repaint(wndlist_t *wndlist)
 static void wndlist_button_clicked(ui_pbutton_t *pbutton, void *arg)
 {
 	wndlist_entry_t *entry = (wndlist_entry_t *)arg;
+	wndmgt_window_info_t *winfo = NULL;
 	sysarg_t dev_id;
+	errno_t rc;
+	bool minimized = false;
 
 	/* ID of device that clicked the button */
 	dev_id = entry->wndlist->ev_idev_id;
 
-	(void) wndmgt_activate_window(entry->wndlist->wndmgt,
-	    dev_id, entry->wnd_id);
+	rc = wndmgt_get_window_info(entry->wndlist->wndmgt,
+	    entry->wnd_id, &winfo);
+	if (rc == EOK) {
+		minimized = (winfo->flags & wndf_minimized) == 0;
+	}
+
+	if (!minimized) {
+		(void) wndmgt_activate_window(entry->wndlist->wndmgt,
+	        dev_id, entry->wnd_id);
+	} else {
```

Comment:
Now here's the big problem. If you click the button of a window that's not minimized and is *not focused*, instead of activating the window it will minimize it. Also note that with multiple seats, each window can have zero, one or more foci. A window should only be minimized if it is the focus of the same seat that clicked the window button.